### PR TITLE
Add ecs info to containerinsight receiver

### PR DESCRIPTION
**Pull request by JohnWu20**
**Description:** <Describe what has changed.>
Fixed the repeat interface define [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/3052086014) and resubmit the pr for this [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3853): 

**Testing:** <Describe what testing was performed and which tests were added.>
Unit test

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by JohnWu20** [comment]
Hi @Aneurysm9 , I checked the exist http client and I think it shouldn’t have any relationship with the http I created because:
1, The http client I created uses retryablehttp to do the error retry logic which actually this http client doesn’t have. (prw exporter implemented the retry logic)
2, The http client I created will check the response size and deal with some corner cases for the http response of calling ECS agent endpoint.
Using the exist  http client can’t meet my requirements.

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+type MockHTTPClient struct {
+	response *http.Response
+	err      error
+}
+
+func (m *MockHTTPClient) Do(reqest *http.Request) (*http.Response, error) {
+	return m.response, m.err
+}
```
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go:68]: This looks identical to `fakeClient`.  That one exists in a test file and isn't exported, so it is preferable to an exported implementation.

```suggestion
```

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+// Check the channel is closed or not.
+func IsClosed(ch <-chan bool) bool {
```
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go:71]: Does this need to be exported?  The smaller the exported surface area the better.

File: internal/aws/containerinsight/go.mod [hunk 3]
```diff
@@ -3,13 +3,8 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/co
 go 1.16
 
 require (
-	github.com/pkg/errors v0.9.1 // indirect
 	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/collector v0.30.0
```
[the remark this answers is not in the document]
Comment (JohnWu20) [reply internal/aws/containerinsight/go.mod:7]: Sure, I will revert this change

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go (hunk 2 above)
Comment (JohnWu20) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go:71]: I seem and I will change it

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go (hunk 1 above)
Comment (JohnWu20) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go:68]: Thanks, I will delete the `fakeClient` and use the `MockHTTPClient` only.

Comment (Aneurysm9) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go:68]: I'm sorry if I was unclear.  I would have kept `fakeClient` and removed `MockHttpClient`.  Alternately, `fakeClient` can be removed, but `MockHttpClient` needs to be renamed to `mockHTTPClient` and moved to `utils_test.go`.

File: internal/aws/containerinsight/go.sum [hunk 4]
```diff
@@ -1,33 +1,469 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+cloud.google.com/go v0.38.0/go.mod h1:990N+gfupTy94rShfmMCWGDn0LpTmnzTp2qbd1dvSRU=
```
Comment (Aneurysm9) [inline internal/aws/containerinsight/go.sum:3]: Run `go mod tidy` in this directory.  These changes should go away.

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go (hunk 1 above)
Comment (JohnWu20) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/utils.go]: Thanks for making it clear, have changed it

File: internal/aws/containerinsight/go.sum (hunk 4 above)
Comment (JohnWu20) [reply internal/aws/containerinsight/go.sum]: Thanks, removed these changes.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**